### ISSUE TITLE
Parelelization of assembiles

### ISSUE THREAD

**Issue by BarbourSmith** [jsxcad/JSxCAD#1044]
Having multiple workers gives a big speed boost to parts of a project which can be computed simultaneously, but often a single assembly ends up hanging up one worker for an extended period of time (tens of minutes in some complex cases).

If we were able to parallelize the process of computing an assembly we could see a dramatic speed increase by using the processor more fully.

**Comment by pentacular** [comment]
Yes, this is a good idea, and I think we could generalize it to booleans in general.

To do this I think we'd need to add a boolean operation protocol, have the worker construct a boolean request, ask the browser to resolve it.

The browser would then dispatch to idle workers, and finally respond to the originating worker.

The challenges here would be in garbage management and the need to construct duplicate cgal meshes.

Let's give it a try.
